env.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import print_function
+
+import argparse
+import glob
+import os
+import sys
+import textwrap
+import subprocess
+
+
+def ensure_dir(path):
+    if os.path.exists(path):
+        return
+    
+    os.mkdir(path)
+    return path
+
+
+def traverse_toolsets(f):
+    msbuild_dir = r'C:\Program Files (x86)\MSBuild'
+    toolsets = glob.glob(r'{}\Microsoft.Cpp\v4.0\*\Platforms\*\PlatformToolsets\*'.format(msbuild_dir))
+    for toolset in toolsets:
+        f(toolset)
+
+
+def clcache_props_path(toolset):
+    return os.path.join(toolset, 'ImportAfter', 'Clcache.props')
+
+
+def generate_props_content(clcache_dir):
+    return textwrap.dedent('''
+        <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+          <PropertyGroup>
+            <ExecutablePath>{clcache_dir};$(ExecutablePath)</ExecutablePath>
+          </PropertyGroup>
+        </Project>''').format(clcache_dir=clcache_dir)[1:]
+
+
+def check_call_quiet(*args, **kwargs):
+    with open(os.devnull, 'w') as devnull:
+        kwargs['stderr'] = devnull
+        kwargs['stdout'] = devnull
+        subprocess.check_call(*args, **kwargs)
+
+
+def set_system_variable(var, value):
+    if value:
+        check_call_quiet(['setx', '-m', var, value])
+    else:
+        reg_path = r'HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\Environment'
+        try:
+            check_call_quiet(['reg', 'query', reg_path, '/V', var])
+        except subprocess.CalledProcessError:
+            # Variable doesn't exists
+            pass
+        else:
+            check_call_quiet(['reg', 'delete', reg_path, '/F', '/V', var])
+    
+        
+def install(exe, cache_dir):
+    def f(toolset):
+        clcache_props = clcache_props_path(toolset)
+        ensure_dir(os.path.dirname(clcache_props))
+        with open(clcache_props, 'w') as f:
+            f.write(generate_props_content(os.path.dirname(exe)))
+    
+    traverse_toolsets(f)
+    set_system_variable('CLCACHE_DIR', cache_dir)
+    set_system_variable('CL', '/MP4')
```

Comment:
Yes, this is doubtful place.
The thing is that in out projects /Fo always point to directory, not to file. And without setting CL variable we always has CallsWithMultipleSourceFiles.
Since I wrote the script in the first place for my team and was not sure of its usefulness for the clcache community, I left this line. But it can easily be removed if need be to get into the upstream.